IdentityCore/src/webview/operations/MSIDBrokerInstallOperation.m

```diff
@@ -0,0 +1,106 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDBrokerInstallOperation.h"
+#import "MSIDWebviewResponse.h"
+#import "MSIDWebWPJResponse.h"
+#import "MSIDBrokerUrlHandlerUtil.h"
+
+#if TARGET_OS_IPHONE
+#import "MSIDAppExtensionUtil.h"
+#endif
+
+@interface MSIDBrokerInstallOperation()
+
+@property (nonatomic) NSURL *appInstallLink;
+
+@end
+
+@implementation MSIDBrokerInstallOperation
+
+- (nullable instancetype)initWithResponse:(MSIDWebviewResponse *)response
+                                    error:(NSError **)error
+{
+    #if TARGET_OS_IPHONE
+    self = [super initWithResponse:response
+                             error:error];
+    if (self)
+    {
+        if (![response isKindOfClass:MSIDWebWPJResponse.class] || [NSString msidIsStringNilOrBlank:[(MSIDWebWPJResponse *)response appInstallLink]])
+        {
+            return nil;
+        }
+        
+        MSIDWebWPJResponse *wpjResponse = (MSIDWebWPJResponse *)response;
+        _appInstallLink = [NSURL URLWithString:wpjResponse.appInstallLink];
+    }
+    
+    return self;
+    #else
+    return nil;
+    #endif
+}
+
+- (void)invokeWithCompletion:(MSIDBaseOperationCompletionHandler)completion
+{
+    completion(NO, nil);
+}
+
+- (void)invokeWithinteractiveTokenRequestParameters:(MSIDInteractiveTokenRequestParameters *)interactiveTokenRequestParameters
+                                         completion:(MSIDBaseOperationCompletionHandler)completion
+{
+    if (!completion)
+    {
+        return;
+    }
+    
+    #if TARGET_OS_IPHONE
+    [MSIDBrokerUrlHandlerUtil UrlImplementation:nil
+                              brokerInstallLink:self.appInstallLink
+                           tokenRequestProvider:nil
+                              requestParameters:nil
+                          interactiveParameters:interactiveTokenRequestParameters
+                          sdkBrokerCapabilities:nil
+                           tokenSpecificHandler:nil
+                                       complete:^(MSIDBrokerHelperErrorType __unused errorType, NSError *error, BOOL __unused shouldFallbackToLocalController) {
+        switch (errorType)
+        {
+            case MSIDBrokerUrlHandlerBase64UrlKeyInvalidError:
```

Comment:
This logic seems duplicate to what is in broker controller.

Reply:
This is a different route later needs to be handled in CPP

Reply:
Yes, let's take this discussion separately offline :)

Reply:
Removed those classes as we discussed so resolve this conversation